legate/driver/config.py

```diff
@@ -176,21 +176,29 @@ class Config:
     """
 
     def __init__(self, argv: ArgList) -> None:
-        self.argv = argv
-
+        user_script = next((x for x in argv if x.endswith(".py")), None)
+        if user_script:
+            user_script_index = argv.index(user_script)
+            legate_argv = argv[:user_script_index]
+            user_argv = argv[user_script_index:][1:]
+        else:
+            legate_argv = argv
+            user_argv = []
+
+        self.argv = legate_argv
+
+        # extra might contain legion command line options for now
         args, extra = parser.parse_known_args(self.argv[1:])
 
         colors.ENABLED = args.color
 
         # only saving this for help with testing
         self._args = args
 
-        self.user_script = next((x for x in extra if x.endswith(".py")), None)
+        self.user_script = user_script
 
-        user_opts = list(extra)
-        if self.user_script in user_opts:
-            user_opts.remove(self.user_script)
-        self.user_opts = tuple(user_opts)
+        # extra here might be legion, realm, or cunumeric command line options
+        self.user_opts = tuple(user_argv) + tuple(extra)
```

Comment:
```suggestion
        self.user_opts = tuple(extra) + tuple(user_argv)
```

Since the `extra` came before `user_argv` in the original command line.